### PR TITLE
fix: limit parallel event queries, goerli block lookback

### PR DESCRIPTION
**Pull request by daywiss**
### Issues
We are seeing RPC provider errors when loading specific proposals that  use osnap. This is due to long lookbacks and many parallel network requests for event data.
image

Fixes #

### Changes 
*_(Briefly describe the changes made in this PR)_*

1. This introduces "bluebird promises", specifically the map function, which allows us to set a concurrency level. this is used in the most likely place where there are too many requests made in parallel using promise.all.
2. we further add deploy block numbers for goerli, in case the users safe is in goerli, previously this was starting a query at block 0.
3. we also fix the default "something went wrong" at the bottom of the transaction form in the proposal.
image

### How to test
*_(Explain how the changes can be tested, including any required setup steps)_*

1. you can see a something went wrong error here: https://snapshot.org/#/frp.eth/proposal/0xfd99f0127621cb5b644114dc6195fc0bbd3da7e88b41cce409b5ed82214ce106
2. see in dev console `ERR_INSUFFICIENT_RESOURCES`
3. visit this same proposal on this build and see that it eventually loads without errors.

### To-Do
this may buy more time to develop a better, faster caching solution

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [ ] I have tested my changes on different screen sizes (sm, md)
- [ ] I have tested my changes on a custom domain

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*

### CODE REVIEW

File: src/plugins/safeSnap/constants.ts [hunk 1]
```diff
@@ -386,3 +386,37 @@ export const MULTI_SEND_VERSIONS: Record<
   [MULTI_SEND_VERSION.V1_2_0]: MULTI_SEND_V1_2_0,
   [MULTI_SEND_VERSION.V1_3_0]: MULTI_SEND_V1_3_0
 };
+
+// to potnetially cut down on event ranges we query, hard code some deploy blocks for contracts
+export type ContractData = {
+  network: string;
```
Comment (daywiss) [inline src/plugins/safeSnap/constants.ts:392]: more robust constants to describe contracts. unfortunately getting deploy blocks are hard to do without hard coding it

File: src/plugins/safeSnap/utils/events.ts [hunk 2]
```diff
@@ -129,14 +130,20 @@ export async function pageEvents<E>(
...
+    ranges.push({
+      start: state.currentStart,
+      end: state.currentEnd,
+      index: index++
+    });
     state = rangeSuccessDescending(state);
   } while (!state.done);
 
-  return (await Promise.all(ranges.map(fetchEvents))).flat();
+  return (await bb.map(ranges, fetchEvents, { concurrency })).flat();
```
Comment (daywiss) [inline src/plugins/safeSnap/utils/events.ts:148]: this is really the main change, this limits promises being executed to `concurrency` amount, by default 5.  performance is still quite good in terms of total loading time.

File: src/plugins/safeSnap/utils/umaModule.ts [hunk 3]
```diff
@@ -138,13 +150,12 @@ export const getModuleDetailsUma = async (
...
-
   const latestBlock = await provider.getBlock('latest');
   // modify this per chain. this should be updated with constants for all chains. start block is og deploy block.
   // this needs to be optimized to reduce loading time, currently takes a long time to parse 3k blocks at a time.
-  const oGstartBlock = network === '1' ? 17167414 : 0;
-  const oOStartBlock = network === '1' ? 16636058 : 0;
-  const maxRange = network === '1' ? 3000 : 10000;
+  const oGstartBlock = getDeployBlock(network, 'OptimisticGovernor');
+  const oOStartBlock = getDeployBlock(network, 'OptimisticOracleV3');
+  const maxRange = network === '1' || network === '5' ? 3000 : 10000;
```
Comment (daywiss) [inline src/plugins/safeSnap/utils/umaModule.ts:158]: include goerli here to limited block ranges. not sure if strictly required, but cant hurt

**Review by bonustrack — APPROVED** [review]
tACK

**Pull request merged**